packages/scoped-elements/src/ScopedElementsMixin.js

```diff
@@ -95,7 +95,18 @@ const ScopedElementsMixinImplementation = superclass =>
         elementStyles,
       } = /** @type {typeof ScopedElementsHost} */ (this.constructor);
 
-      if (!this.registry) {
+      // @ts-ignore
+      const createRegistry =
+        !this.registry ||
+        (!Object.prototype.hasOwnProperty.call(this.constructor, '__registry') &&
```

Comment:
I would change the order to break the circuit earlier when the regitry is atteched to the instance instead of the class

```js
(this.registry === this.constructor.__registry && !Object.prototype.hasOwnProperty.call(this.constructor, '__registry'));
```

Reply:
Done